libs/datamodel/core/src/transform/ast_to_dml/validate.rs

```diff
@@ -271,6 +271,98 @@ impl<'a> Validator<'a> {
         }
     }
 
+    fn validate_enum_default_values(
+        &self,
+        data_model: &dml::Datamodel,
+        ast_model: &ast::Model,
+        model: &dml::Model,
+    ) -> Result<(), ErrorCollection> {
+        let mut errors = ErrorCollection::new();
+
+        for field in model.scalar_fields() {
+            if let Some(DefaultValue::Single(PrismaValue::Enum(enum_value))) = &field.default_value {
+                if let FieldType::Enum(enum_name) = &field.field_type {
+                    if let Some(dml_enum) = data_model.find_enum(&enum_name) {
+                        if !dml_enum.values.iter().map(|v| &v.name).any(|value| value == enum_value) {
+                            errors.push(DatamodelError::new_directive_validation_error(
+                                &format!(
+                                "{}",
+                                "The defined defaultvalue is not a valid value of the enum specified for the field."
+                            ),
+                                "default",
+                                ast_model.find_field(&field.name).span,
+                            ))
+                        }
+                    }
+                }
+            }
+        }
+
+        if errors.has_errors() {
+            Err(errors)
+        } else {
+            Ok(())
+        }
+    }
+
+    fn validate_auto_increment(&self, ast_model: &ast::Model, model: &dml::Model) -> Result<(), ErrorCollection> {
+        let mut errors = ErrorCollection::new();
+
+        if let Some(data_source) = self.source {
+            if !data_source.combined_connector.supports_multiple_auto_increment()
+                && model.auto_increment_fields().count() > 1
+            {
+                errors.push(DatamodelError::new_directive_validation_error(
+                    &format!(
+                        "{}",
+                        "The `autoincrement()` default value is used twice on this model even though the underlying database only allows one instance per table."
+                    ),
+                    "default",
+                    ast_model.span,
+                ))
+            }
+
+            // go over all fields
+            for field in model.scalar_fields() {
+                let ast_field = ast_model.find_field(&field.name);
+
+                if !field.is_id
+                    && field.is_auto_increment()
+                    && !data_source.combined_connector.supports_non_id_auto_increment()
+                {
+                    errors.push(DatamodelError::new_directive_validation_error(
+                    &format!(
+                        "{}",
+                        "The `autoincrement()` default value is used on a non-id field even though the database does not allow this."
```

Comment:
We should stick to our Prisma language and call it a `datasource` imo. And i would substitute `allow` with `support` which sounds more like a given fact to me.